clang/include/clang/Sema/Attr.h

```diff
@@ -123,6 +123,13 @@ inline bool isInstanceMethod(const Decl *D) {
   return false;
 }
 
+inline bool hasImplicitObjectParameter(const Decl *D) {
+  if (const auto *MethodDecl = dyn_cast<CXXMethodDecl>(D))
+    return MethodDecl->isInstance() &&
+           !MethodDecl->hasCXXExplicitFunctionObjectParameter();
+  return false;
+}
```

Comment:
```suggestion
inline bool hasImplicitObjectParameter(const Decl *D) {
  if (const auto *MethodDecl = dyn_cast<CXXMethodDecl>(D))
    return MethodDecl->isImplicitObjectMemberFunction();
  return false;
}
```

Reply:
A method can be static and not have the implicit this available, right? I think we still should check `MethodDecl->isInstance()`

Reply:
> A method can be static and not have the implicit this available, right? I think we still should check `MethodDecl->isInstance()`

No, `isImplicitObjectMemberFunction()` already checks for that:
```c++
bool CXXMethodDecl::isImplicitObjectMemberFunction() const {
  return !isStatic() && !hasCXXExplicitFunctionObjectParameter();
}
```

Reply:
Also, yes, I just accidentally edited someone’s comment *again*